### PR TITLE
Examples for the sliding operator 

### PR DESCRIPTION
**Pull request by johanandren**
Refs #25468

### CODE REVIEW

**Comment by johanandren** [comment]
Crap, forgot the signature for Scala + Java

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md [hunk 1]
```diff
@@ -4,20 +4,47 @@ Provide a sliding window over the incoming stream and pass the windows as groups
...
+
+One use case for sliding is to implement a moving average, here we do that with a "period" of `5`:
+
+Scala
+:   @@snip [Sliding.scala](/akka-docs/src/test/scala/docs/stream/operators/sourceorflow/Sliding.scala) { #moving-average }
+
+Java
+:   @@snip [Sliding.java](/akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Sliding.java) { #moving-average }
+
+Sliding can also be used to do simple windowing, see @ref[splitAfter](splitAfter.md).
```
Comment (johanandren) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md]: Will be there when #28480 is merged :)

**Review by chbatey — APPROVED** [review]
Looks good

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md [hunk 2]
```diff
@@ -4,20 +4,47 @@ Provide a sliding window over the incoming stream and pass the windows as groups
...
 @ref[Simple operators](../index.md#simple-operators)
 
-@@@div { .group-scala }
-
 ## Signature
 
-@@signature [Flow.scala](/akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala) { #sliding }
-
-@@@
+@apidoc[Flow.sliding](Flow) { scala="#sliding(n:Int,step:Int):FlowOps.this.Repr[scala.collection.immutable.Seq[Out]]" java="#sliding(int,int)" }
```
Comment (chbatey) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md]: How do you work these out easily? Is it the same as the link in the API doc?

Comment (patriknw) [reply akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md]: I have same question. This approach seems error prone and difficult for contribitors and us to figure out. When I originally proposed linking to apidoc the idea was that users would have to go to the apidocs to see the signature. Link would be something like Flow#sliding

Comment (chbatey) [reply akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md]: So it is quite straightforward as you go to the api doc and copy the #xxx bit from the URL.

We can add something to the issue/contributing docs for how to do this

It still only produces a link to the API doc.

Comment (johanandren) [reply akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md]: Link looks like that `Flow.sliding` the tricky part is just there to make it lead to the right section in the Scaladoc vs the Javadocs. It's the anchor part so I "just" go to to the respective API:docs and copy that.

Comment (johanandren) [reply akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md]: I agree it is suboptimal, if someone has a better solution I'm all ears. 👂👂👂👂

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md [hunk 3]
```diff
@@ -4,20 +4,47 @@ Provide a sliding window over the incoming stream and pass the windows as groups
...
 ## Description
 
 Provide a sliding window over the incoming stream and pass the windows as groups of elements downstream.
 
 Note: the last window might be smaller than the requested size due to end of stream.
 
+## Examples
+
+In this first sample we just see the behavior of the operator itself, first with a window of 2 elements and @scala[the default
+`step` which is 1]@java[a step value of 1].
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/sliding.md]:
```suggestion
`step`, which is '1']@java[a step value of '1'].
```

**Review by raboof — APPROVED** [review]
Nice simple examples. One optional tweak suggestion.

**Pull request merged**